samples/Common.c

```diff
@@ -693,6 +693,7 @@ STATUS createSampleConfiguration(PCHAR channelName, SIGNALING_CHANNEL_ROLE_TYPE
 
 #ifdef IOT_CORE_ENABLE_CREDENTIALS
```

Comment:
I really don't understand why it was decided to make this a compile time define. We spent so much effort to avoid any compile time definitions in the headers and especially in the source files. The whole reason for the platform-level integration in CommonDefs and the way the entire source tree is architected is to avoid as much as possible the conditional compilation which is terrible when it comes to readability and maintenance of the codebase.

Reply:
Yes, I am also a little confused here. I will provide the sample code for ecs credential provider later, and I do not know do I need to add one more definition?